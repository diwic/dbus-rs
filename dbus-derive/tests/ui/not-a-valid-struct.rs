use dbus_macros::DbusStruct;

#[derive(DbusStruct)]
pub enum Arg {
    Opt1,
    Opt2,
}

#[derive(DbusStruct)]
pub struct Arg3;

fn main() {}
