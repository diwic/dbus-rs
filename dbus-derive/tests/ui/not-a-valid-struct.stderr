error: Unsupported shape `enum`. Expected struct with named fields or unnamed fields.
 --> tests/ui/not-a-valid-struct.rs:4:5
  |
4 | pub enum Arg {
  |     ^^^^

error: Unsupported shape `no fields`. Expected named fields or unnamed fields.
  --> tests/ui/not-a-valid-struct.rs:10:5
   |
10 | pub struct Arg3;
   |     ^^^^^^
